In [1]:
import pandas as pd
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt
import seaborn as sns


In [3]:
data = pd.read_csv('machine1.csv')
data

New machine  Old machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

# One tailed t-test
ttest & p_value analysisi


In [18]:
#data1=data['New machine'].mean() 
#data1
np.mean(data['Old machine'])

43.230000000000004

In [22]:
old_mean=round(np.mean(data['Old machine']),3)
st.ttest_1samp(data['New machine'],old_mean)

Ttest_1sampResult(statistic=-5.043318535038297, pvalue=0.0006966376076759338)

In [24]:
print('p value (single tailed): ',st.ttest_1samp(data['New machine'],43.23)[1]/2)


p value (single tailed):  0.0003483188038379669


conclusion: 
1. p is low and we can reject H0
2. statistic=-5.043318535038297 is nagative so we keep old machine

# Matched Pairs Test

data = pd.read_csv('pokemon.csv')
data

In [29]:
pokemon = data[["Attack", "Defense"]]
pokemon.head()

Attack  Defense
0      49       49
1      62       63
2      82       83
3     100      123
4      52       43

In [28]:
st.ttest_rel(data['Attack'], data['Defense'])

Ttest_relResult(statistic=4.325566393330478, pvalue=1.7140303479358558e-05)

conclusion:
    
1. H0 is the hypothesis that the defense and attack scores are equal.
However, statistic is 4.325566393330478 that is positive. P value is 1.714 and it is positive too.It means the hypothesis is wrong and  defense and attack scores are not equal.
2. We reject H0 and keep H1.
